### ISSUE TITLE
Fasta signals warnings, not errors, on uses of void

### ISSUE THREAD

**Issue by ngolovleva** [dart-lang/sdk#32996]
The test below is successfully executed with dart, while analyzer in _--strong_ and _--preview-dart-2_ modes reports errors.
```dart
class B {
  static void staVoid = null;
  static Object oSta = new Object();
  void insVoid = null;
  Object oIns = new Object();
}

void gloVoid = null;
Object oGlo = new Object();

void checkGlo() {
  oGlo = gloVoid;
}

void checkIns() {
  B a1 = new B();
  a1.oIns = a1.insVoid;    // dartanalyzer -> error
}

void checkSta() {
  B.oSta = B.staVoid;    // dartanalyzer -> error
}

dynamic checkLoc() {
  void locVoid = null;
  Object oLoc = new Object();
  oLoc = locVoid;
  return oLoc;
}

main() {
  checkGlo();
  checkIns();
  checkSta();
  checkLoc();
}
```
```
dartanalyzer --preview-dart-2 test.dart 
Analyzing test.dart...
  error • The expression here has a type of 'void', and therefore cannot be used at test.dart:13:10 • use_of_void_result
  error • The expression here has a type of 'void', and therefore cannot be used at test.dart:20:13 • use_of_void_result
  error • The expression here has a type of 'void', and therefore cannot be used at test.dart:26:12 • use_of_void_result
  error • The expression here has a type of 'void', and therefore cannot be used at test.dart:34:10 • use_of_void_result
4 errors found.
```
The same behaviour is observed if instead of type **Object** type **dynamic** is used.
Checked with Dart VM version: 2.0.0-dev.50.0 (Wed Apr 25 11:29:49 2018 +0200) on "linux_x64"

**Comment by bwilkerson** [comment]
@leafpetersen @lrhn Which, if either, is correct?

**Comment by eernstg** [comment]
I couldn't match up the line numbers in the dartanalyzer output with the source code (or the '-> error' comments), but we should get the following errors (indicated using test style comments):
```dart
class B {
  static void staVoid = null;
  static Object oSta = new Object();
  void insVoid = null;
  Object oIns = new Object();
}

void gloVoid = null;
Object oGlo = new Object();

void checkGlo() {
  oGlo = gloVoid; //# 01: compile-time error
}

void checkIns() {
  B a1 = new B();
  a1.oIns = a1.insVoid; //# 02: compile-time error
}

void checkSta() {
  B.oSta = B.staVoid; //# 03: compile-time error
}

dynamic checkLoc() {
  void locVoid = null;
  Object oLoc = new Object();
  oLoc = locVoid; //# 04: compile-time error
  return oLoc;
}

main() {
  checkGlo();
  checkIns();
  checkSta();
  checkLoc();
}
```
The reason for each of these errors is that an expression of static type `void` is evaluated and the result is neither discarded nor passed on in any of the ways on the "positive list" (and everything else than these exceptions is an error).

**Comment by natebosch** [comment]
in `2.1.0-dev.0.0` these show up as Warnings. Should they be turned in to Errors?

**Comment by eernstg** [comment]
Yes, they should be errors.

**Comment by kmillikin** [comment]
Fixed in 9bdf2489b54841567005eaff7033f3451a047b37.
